src/dmd/dsymbol.d

```diff
@@ -793,6 +793,17 @@ extern (C++) class Dsymbol : ASTNode
                         }
                     }
                 }
+                if (auto ad = isAliasDeclaration())
+                {
+                    if (auto tag = ad.getType().isTypeTag())
+                    {
+                        // New typedef of a struct that has already been defined.
+                        // Resolve the type to the original declaration.
+                        import dmd.typesem : typeSemantic;
+                        tag.typeSemantic(loc, sc);
```

Comment:
can we not do this in semantic rather than in dsymbolsem? I'd rather avoid having the AST dependant on semantic

Reply:
The collision is detected in the `importAll` phase of the compilation.

Really we should either have a new symbol table just for the C tag namespace, or add a new field to `Dsymbol` to chain together all collisions that aren't an actually a redeclaration.

This will need to be dealt with at some point to allow `struct foo;` and `int foo()` to be declared in the same scope.  For now, just dealing with the easy shortcomings to weed out the [next hidden issue](https://github.com/ibuclaw/importC/) in the importC implementation.